src/adapters/github/storage-layer.ts

```diff
@@ -0,0 +1,342 @@
+     * "delete" will remove the session, this will break things without a new session.
+     */
+    async handleSession<TAction extends "create" | "delete">(session: string, action: TAction) {
+        const dbObject = await this.retrieveStorageDataObject("session");
+
+        if (action === "create") {
+            dbObject.session = session;
+        } else {
+            dbObject.session = "";
+        }
+
+        return await this.storeData(dbObject);
+    }
+
+    /**
+     * Updates the user bank with the new user data. If the user already exists,
+     * it will overwrite the existing data or delete it if the action is "delete".
+     * 
+     * This is a Telegram focused method, so we'll use the telegramId as the key.
+     * This way notifications can only be received to the account which is subscribing,
+     * them may choose to listen into other users, but cannot enable notifications for
+     * another user.
+     */
+    async handleUserBank<TType extends "create" | "delete" | "update">(user: UserBank[string], action: TType) {
+        const dbObject = await this.retrieveStorageDataObject("userBank");
+
+        const existingUser = dbObject[user.telegramId];
+
+        if (action === "create" && existingUser || action === "delete" && !existingUser) {
+            throw new Error("User already exists or does not exist");
+        }
+
+        if (action === "create" || action === "update") {
+            dbObject[user.telegramId] = user;
+        } else {
+            delete dbObject[user.telegramId];
+        }
+
+        return await this.storeData(dbObject);
+    }
+
+    // Low level fetching and storage
+
+    /**
+     * Stores data in the GitHub repo. It will overwrite the existing
+     * data, so ensure you're passing the full object, not just the
+     * updated properties, or mistakes will be made.
+     * 
+     * TODO: safety check to ensure we are not accientally deleting data.
+     */
+    async storeData<TType extends StorageTypes>(data: RetrievalHelper<TType>) {
+        let path;
+        let type: StorageTypes;
+        const { sha } = data;
+
+        data = deleteAllShas(data);
+
+        if (isChatsStorage(data)) {
+            path = this.chatStoragePath;
+            type = "allChats";
+        } else if (isUserBankStorage(data)) {
+            path = this.userBankPath;
+            type = "userBank";
+        } else if (isSingleChatStorage(data)) {
+            path = this.chatStoragePath;
+            type = "singleChat";
+        } else if (isSessionStorage(data)) {
+            path = this.telegramSessionPath;
+            type = "session";
+        } else {
+            throw new Error("Invalid data type");
+        }
+
+        const content = JSON.stringify(data, null, 2);
+
+        try {
+            await this.octokit.repos.createOrUpdateFileContents({
+                owner: this.owner,
+                repo: this.repo,
+                path,
+                message: `Update ${type} data`,
```

Comment:
I wonder if it is a better idea to do git storage on another branch. 

Then we don't need to boost our commit count on our default branch. 

Also this should follow conventional commits format.

Reply:
Yeah that might be a good idea

What are your thoughts on having a dedicated app/org access token per partner for there storage needs as opposed to (if possible) using the kernel' authorized token?

I imagine once rolled out completely this will eat quite a large chunk of a token' rate limit due to always having to have the most up to date commit reference etc. I expect a lot of redundant-yet-necessary calls to be made.

> Also this should follow conventional commits format.

Cool, just my war-cry, "chore:" I guess yeah

Reply:
I targeted the branch "storage"